lib/model/tank.dart

```diff
@@ -0,0 +1,21 @@
+class Tank<T1, T2> {
```

Comment:
Why is this a templated class? It's T1 always a String? Isn't T2 always a string? Could the name and IP default to empty strings?